### PR TITLE
externals: allow users to use system cpp-httplib

### PR DESCRIPTION
**Pull request by Castor215**
Continuing the work of #6501.

- - -
This change is [Reviewable](https://reviewable.io/reviews/citra-emu/citra/7034)

### CODE REVIEW

**Review by Steveice10 — APPROVED** [review]

File: externals/cmake-modules/FindCppHttp.cmake [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+endif()
+
+if(CppHttp_FOUND AND NOT TARGET cpp-httplib::cpp-httplib)
+	add_library(cpp-httplib::cpp-httplib INTERFACE IMPORTED)
+	set_target_properties(cpp-httplib::cpp-httplib PROPERTIES
+		INTERFACE_INCLUDE_DIRECTORIES "${CPP-HTTP_INCLUDE_DIR}"
+		INTERFACE_LINK_LIBRARIES "${CPP-HTTP_LIBRARIES}"
+		IMPORTED_LOCATION "${CPP-HTTP_LIBRARIES}"
+	)
+endif()
```
Comment (FearlessTobi) [inline externals/cmake-modules/FindCppHttp.cmake:35]: nit: missing a newline here

File: externals/CMakeLists.txt [hunk 2]
```diff
@@ -199,7 +199,12 @@ endif()
 
 # httplib
 add_library(httplib INTERFACE)
-target_include_directories(httplib INTERFACE ./httplib)
+if(USE_SYSTEM_CPP_HTTPLIB)
+	find_package(CppHttp REQUIRED)
```
Comment (FearlessTobi) [inline externals/CMakeLists.txt:203]: nit: we use spaces and not tabs for formatting

**Review by FearlessTobi — APPROVED** [review]
LGTM, just two small things.

File: externals/CMakeLists.txt (hunk 2 above)
Comment (Castor215) [reply externals/CMakeLists.txt]: Oops. This is what I get for using mousepad instead of my usual IDE. I have pushed the suggested changes.

**Comment by PabloMK7** [comment]
NOTE: Make sure to use HTTPLIB 0.14.1 or posterior.

**Comment by Castor215** [comment]
> NOTE: Make sure to use HTTPLIB 0.14.1 or posterior for incoming HTTP:C changes.
This might be a problem for me, as 14.1 is not yet in the Debian repositories. I could make the version check, but I would then be unable to test it. @Steveice10 or @FearlessTobi please advise.

**Pull request merged**